src/security/cli_security.c

```diff
@@ -145,7 +146,16 @@ get_cred_from_response(Drpc__Response *response, d_iov_t *cred)
 		D_GOTO(out, rc = -DER_PROTO);
 	}
 
+	if (cred_resp->cred->verifier == NULL) {
+		D_ERROR("Credential did not include verifier\n");
+		D_GOTO(out, rc = -DER_PROTO);
+	}
+
 	rc = auth_cred_to_iov(cred_resp->cred, cred);
+
+	/* If present clear out the verifier (the secret part) */
+	verifier = cred_resp->cred->verifier;
+	memzero_explicit(verifier->data.data, verifier->data.len);
```

Comment:
Is the verifier->data.len variable or can we count on it being a fixed size?  If it's fixed, it would be better to use a constant to avoid memsetting off the boundaries of the buffer.  If that's not possible, what you have is fine.  It just worries me that there's potential for an overrun.

Reply:
So the length of the data can be variable I believe. The reason we use these two fields is because the protobuf is the only canonical source for what the allocation of that pointer and its length is.  If for some odd reason we assumed that the verifier was a certain length but it packed to a different length because of a change in protobuf internals we could have an issue. This way we know that the protobuf is at least internally consistant.